### ISSUE TITLE
Loss for Refinement Module

### ISSUE THREAD

**Issue by qchenclaire** [foolwood/SiamMask#14]
If training with the refinement module, what is the loss for mask you use? a) Eq. 3 in the paper; b) Eq.3 plus something like binary cross entropy for the final mask;  c)binary cross entropy for the final mask?

**Comment by qchenclaire** [comment]
Also if I understand correctly, in Eq.3, n ranges from 0 to 17x17-1 and you set y_n in the same way a proposal is labeled as positive/negative in faster rcnn, i.e. iou > 0.5? We need to do ROI align on the ground truth mask to get c_n. The boxes for ROI align are the predicted boxes from box regression? And i, j respectively ranges from 0 to 126 if we decide to unflatten RoW and resize it to 127x127? This has a very similar flavor to MaskRCNN.
BTW, there seems to be a bug here https://github.com/foolwood/SiamMask/blob/master/tools/test.py#L257 if refine_enable is set False. We need to do bilinear sampling here instead of just view()
